src/sentry/api/serializers/models/group.py

```diff
@@ -571,26 +577,55 @@ def _get_subscriptions(
         if not groups:
             return {}
 
+        notification_settings, query_groups = None, None
         groups_by_project = collect_groups_by_project(groups)
-        notification_settings_by_scope = transform_to_notification_settings_by_scope(
-            notifications_service.get_settings_for_user_by_projects(
-                type=NotificationSettingTypes.WORKFLOW,
-                user_id=user.id,
-                parent_ids=list(groups_by_project.keys()),
+        if should_use_notifications_v2(groups[0].project.organization):
+            notification_settings = {
+                NotificationSettingEnum(key): NotificationSettingsOptionEnum(val)
+                for key, val in notifications_service.get_setting_options_for_user(
+                    user_id=user.id,
+                    project_ids=list(groups_by_project.keys()),
+                    type=NotificationSettingEnum.WORKFLOW,
+                ).items()
+            }
+            query_groups = {
```

Comment:
@snigdhas I don't think there is much harm in querying all the users to make this simpler. The number of users who opt out is likely going to be small. Would prefer cleaner code if possible.